lib/core/src/sdk.rs

```diff
@@ -869,18 +913,24 @@ impl LiquidSdk {
                     .await
             }
             SendDestination::Bolt11 { invoice } => {
-                self.pay_invoice(&invoice.bolt11, *fees_sat).await
+                self.pay_bolt11_invoice(&invoice.bolt11, *fees_sat).await
+            }
+            SendDestination::Bolt12 { offer } => {
+                // TODO Amount hardcoded to 1_000 because we don't yet have it in the request
+                let bolt12_invoice = self.swapper.get_bolt12_invoice(offer, 1_000)?;
```

Comment:
Perhaps it is better to not fetch again the bolt12? Just pass it in the prepare response?

Reply:
I am not sure now we need to fetch it in the prepare but I think we should only fetch either here or in the prepare but not both.

Reply:
You're right. As discussed, removed invoice fetching from prepare.